apps/dialtone-documentation/docs/.vuepress/baseComponents/CodeExampleTabs.vue

```diff
@@ -57,26 +57,67 @@
 </template>
 
 <script setup>
+import { computed, onMounted, ref, nextTick } from 'vue';
 import Prism from 'prismjs';
+import pretty from 'pretty';
 import CopyButton from './CopyButton.vue';
-import { ref } from 'vue';
 import { getUniqueString } from '@workspaceRoot/common/utils';
 
 const props = defineProps({
+  /**
+   * Retrieves the reference to the example component to get the HTML code.
+   * If not provided, the HTML code will be used.
+   *
+   * @returns {Component} The reference to the component.
+   */
+  getComponentRef: {
```

Comment:
Instead of creating this new prop within this component, can we just leave this component with the htmlCode prop as it was before and create a utility function (or composable maybe?) that could convert a vue component to a string of formatted code, and then pass that string into the vueCode prop? I think this would solve some of the separation concerns @iropolo had mentioned.

Reply:
Updated so that htmlCode prop can be a string or a function to get the ref.